frontend/src/lib/components/PropertiesTable/PropertiesTable.tsx

```diff
@@ -195,6 +196,11 @@ interface PropertiesTableType extends BasePropertyType {
     tableProps?: Partial<LemonTableProps<Record<string, any>>>
     highlightedKeys?: string[]
     type: PropertyDefinitionType
+    /**
+     * The container for these properties e.g. the event name of the event the properties are on
+     * Can be used for e.g. to promote particular properties when sorting the properties
+     */
+    parent?: string
```

Comment:
it now needs to know its context so it can detect what properties can be promoted

i couldn't think of a better name than `parent` but I'm open to suggestions

Reply:
`parentEvent` could work, as this is an event? At least it's a bit more descriptive

Reply:
it could be other things in future... e.g. groups have properties 

i played with `container`, `context`, `parent`

i guess it's easy to change and as we add things it might become clear what it should be called